src/battle.py

```diff
@@ -130,6 +140,76 @@ def handle_melee(self, d, best_material, best_enemy_material, stats, enemy_stats
 
             attack = self.get_melee_attack(best_material)
             defense = self.target.get_melee_defense(best_enemy_material)
+            #other_attack = self.get_sidearm_attack(best_material)
+
+            target_is_shield = self.target.weapons[1].shield
+            target_is_ranged = self.target.ranged
+            target_off_handed = self.target.weapons[0].name != self.target.weapons[1].name
+            target_defense_bonus = (target_off_handed) and (not target_is_ranged) and (not target_is_shield)
+
+            armor_pierce = self.get_melee_weapon().armor_pierce
+            is_shield = self.weapons[1].shield
+            is_heavy = self.target.armor.heavy                     
+            is_two_handed = self.weapons[0].name == self.weapons[1].name 
+
+            # if not is_shield and not is_two_handed:
```

Comment:
Why is this whole section commented out? Probably should just remove if it's not important.